### ISSUE TITLE
[duckdb] Numeric operators

### ISSUE THREAD

**Issue by paultiq** [sqlfluff/sqlfluff#7472]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Several numeric operators will not parse: ** for exponentiation, ! for factorial and @ for abs()

Reference: https://duckdb.org/docs/stable/sql/functions/numeric

### Expected Behaviour

The following query is valid duckdb SQL and should parse

### Observed Behaviour

L:   9 | P:   9 |  PRS | Line 9, Position 9: Found unparsable section: '** 3 AS
                       | exponentiation_1,\n      2 ^ 3 AS...'

### How to reproduce

```sql

SELECT
      -- Basic operators
      1 + 2 AS addition,
      5 - 3 AS subtraction,
      4 * 3 AS multiplication,
      10 / 3 AS float_division,
      10 // 3 AS integer_division,
      10 % 3 AS modulo,
      2 ** 3 AS exponentiation_1,
      2 ^ 3 AS exponentiation_2,

      -- Bitwise operators
      5 & 3 AS bitwise_and,
      5 | 3 AS bitwise_or,
      1 << 4 AS shift_left,
      16 >> 2 AS shift_right,
      ~5 AS bitwise_negation,
      5! AS factorial_op,

      @(-42) AS abs_alias,
```

sqlfluff lint test.sql

### Dialect

duckdb

### Version

4.0.4

### Configuration

[tool.sqlfluff.core]
dialect = "duckdb"
templater = "jinja"

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
